datahub-web-react/src/app/tags/TagsTable.tsx

```diff
@@ -62,6 +78,42 @@ const TagsTable = ({ searchQuery, searchData, loading: propLoading, networkStatu
 
     const isLoading = propLoading || networkStatus === NetworkStatus.refetch;
 
+    // Function to delete a tag with confirmation
+    const deleteTag = useCallback(
+        (tagUrn: string) => {
+            const fullDisplayName = entityRegistry.getDisplayName(EntityType.Tag, { urn: tagUrn } as any);
```

Comment:
const fullDisplayName = entityRegistry.getDisplayName(EntityType.Tag, { urn: tagUrn } as any);

don't do this - we should pass in the actual tag entity here.

const fullDisplayName = entityRegistry.getDisplayName(EntityType.Tag, tag);

You can get the tag entity from tagsData. Inside one object in there you'll have an entity field.

tagsData.map(tagData => tagData.entity // This is the object you want to pass in)